Teleport Identity Security - Add Identity Activity Center alerts documentation (#59714)
gravitational/teleport: docs/pages/identity-security/usage/alerts.mdx

---
title: Identity Activity Center Alerts
sidebar_label: Alerts
description: Describes the pre-built security detections that trigger alerts in Teleport Identity Activity Center.
tags:
 - identity-security
---

> Teleport Identity Security provides 50 pre-built security detections that automatically create alerts for suspicious identity-related activities across your infrastructure. These detections monitor events from Teleport and integrated services like AWS, GitHub, and Okta to identify potential security risks.

Reviewer: Do we need to call out the specific number of detections? I don't think knowing the exact number adds any value, and it is likely to go out of date as we refine them in the future.

Teleport Identity Security alerts help you detect and respond to security threats by monitoring for:
- Unusual authentication patterns
- Privilege escalations
- Configuration changes that affect security
- Account compromises
- Policy violations

All detections are pre-configured and currently cannot be modified. The severity levels (Critical, High, Medium, Low) are defined by the Identity Security team based on the potential impact of each detection.

![Alerts](../../../img/access-graph/alerts.png)

> ## Detection categories

Reviewer: ```suggestion
## Detection categories

```

Each category below represents an integrated service where Teleport monitors for security events and suspicious activities pulled from various integration audit logs. The detections within each category are further organized by severity level to help you prioritize your security response efforts.

### AWS detections

> Teleport Identity Security monitors your AWS infrastructure for critical security events:

Reviewer: - I would end the sentence with a `.`, not a `:`
- Why do we only mention "critical security events" here, but then enumerate a bunch of other (non-critical) detections below?

Author: Clarified

#### Critical and high severity
- Root account activities (console login, access key creation, general activity)
- Security settings deletions (CloudTrail, GuardDuty, flow logs)
- EBS encryption disabled
- DB snapshots made public
- Exposed credential policy modifications
- Login profile updates

#### Medium and low severity
- IAM user creation
- DB snapshot attribute modifications
- AWS key deletion or disabling

> Set up AWS detections: [AWS Settings](../integrations/aws-sync.mdx)

Reviewer: The placement of this link makes it read as if this setup is only required for the medium and low severity detections. Recommend moving it up to the intro section.

### GitHub detections

Monitor your GitHub organizations and repositories for security-relevant changes:

#### Critical and high severity
- Organization security updates (SAML, MFA, OAuth restrictions)
- Protected branch policy overrides or deletions
- Repository access and visibility changes
- Secret scanning alerts
- Organization member updates

#### Medium and low severity
- Repository member updates
- Organization moderator additions
- Integration installations

#### Advanced Security Detections
The system includes 26 sub-types of GitHub advanced security change detections that monitor for modifications to security features like Dependabot alerts.

> Set up GitHub detections: [GitHub Settings](../integrations/github.mdx)

Reviewer: This feels like an incomplete thought. Is this a prerequisite for the GitHub Advanced Security detections, or for all GitHub detections?

### Okta detections

Track identity and access management events in your Okta environment:

#### High severity
- Admin MFA disabled
- All MFA factors reset
- OAuth token reuse patterns
- Rate limit violations
- Sign-on policy evaluations
- IDP lifecycle changes
- API token creation/revocation
- Security threat events

#### Medium severity
- Excessive MFA failures
- Multiple password reset attempts
- Dormant account access
- Support-initiated password or MFA resets

### Teleport detections

Monitor Teleport-specific security events:

#### High severity
- Root SSH session initiation
- Authentication without MFA for local accounts
- Unusual authentication failure patterns
- Role creation/updates/deletions
- Connector updates
- Unusual session commands

Set up Teleport detections: [Teleport Settings](../integrations/teleport.mdx)

### Cross-platform detections

#### Impossible travel detection (GitHub, Okta, Teleport)

> - Severity: High

Reviewer: We should be consistent with how we indicate severity.

In previous sections, we had a separate section header for critical, high, medium, etc.

In this section we're listing the severity as a bullet point.

Author: Good feedback, fixed in a commit

- Investigate related events - View associated events, graph and context in the Investigate view, allowing you to understand the full scope of the security incident and trace related activities across your infrastructure.

## Frequently asked questions

### Are these detections available in Teleport Cloud?
No, Identity Activity Center detections are only available in self-hosted Teleport Enterprise deployments. They will be coming to Teleport Enterprise Cloud in Q4 2025.

### Can I write my own detections?
No, detections are currently pre-configured per integration and cannot be customized.

### Can I resolve an alert?
Not currently, but we plan to add workflows to resolve, acknowledge, and mute alert types in future updates.

### What determines the severity level?
Severity levels are defined by the Identity Security Team based on the potential security impact. Feedback on severity assignments is welcome.

### Can I forward these events to another service?
This feature is planned for future release.

### How can I request new detection types?

> For new detection requests based on customer needs, reach out to the Identity Security Team. Customer feedback is welcomed and helps prioritize new detection development.

Reviewer: How do we expect the reader to know how to get in touch with the Identity Security team?

## How do I deploy Identity Security Alerts?
Teleport Identity Security is a separately licensed product available to Teleport Enterprise customers. Alerts and Investigate view are only available in AWS self-hosted deployments. To deploy Identity Security, follow the instructions in [Self-Host Teleport Security - Identity Activity Center](../access-graph/identity-activity-center.mdx).

## Next steps

> - Navigate to **Identity Security** > **Alerts** in the Teleport Web UI to view active alerts

Reviewer: This feels like something that could be mentioned at the very beginning of the document ("Identity Activity Center alerts can be accessed from the ...") instead of as a next step (which is typically links to further reading or related topics)

---
title: Identity Activity Center Alerts
sidebar_label: Alerts
description: Describes the pre-built security detections that trigger alerts in Teleport Identity Activity Center.
tags:
 - identity-security
 - alerts
 - detections
---

# Identity Activity Center Alerts

The Identity Activity Center provides 50 pre-built security detections that automatically create alerts for suspicious identity-related activities across your infrastructure. These detections monitor events from Teleport and integrated services like AWS, GitHub, and Okta to identify potential security risks.

## Overview

Identity Activity Center alerts help you detect and respond to security threats by monitoring for:
- Unusual authentication patterns
- Privilege escalations
- Configuration changes that affect security
- Account compromises
- Policy violations

> All detections are pre-configured and currently cannot be modified. The severity levels (Critical, High, Medium, Low) are defined by the Identity Security Team based on the potential impact of each detection.

Reviewer 2: ```suggestion
All detections are pre-configured and currently cannot be modified. The severity levels (Critical, High, Medium, Low) are defined by the Identity Security team based on the potential impact of each detection.
```
"team" is a common noun here

![Alerts](../../../img/access-graph/alerts.png)

> ## Detection Categories

Reviewer 2: ```suggestion
## Detection categories
```
We use sentence case in headings.

> ### AWS Detections

Reviewer 2: ```suggestion
### AWS detections
```
Sentence case in headings

> Identity Activity Center monitors your AWS infrastructure for critical security events:

Reviewer 2: ```suggestion
Teleport Identity Security monitors your AWS infrastructure for critical security events:
```

> **Critical & High Severity:**

Reviewer 2: ```suggestion
**Critical and high severity:**
```

- Root account activities (console login, access key creation, general activity)
- Security settings deletions (CloudTrail, GuardDuty, flow logs)
- EBS encryption disabled
- DB snapshots made public
- Exposed credential policy modifications
- Login profile updates

> **Medium & Low Severity:**

Reviewer 2: ```suggestion
**Medium and low severity:**
```
Since this is regular prose, rather than a product name, I would use "and" instead of an ampersand for consistency.

- IAM user creation
- DB snapshot attribute modifications
- AWS key deletion or disabling

> ### GitHub Detections

Reviewer 2: ```suggestion
### GitHub detections
```

Monitor your GitHub organizations and repositories for security-relevant changes:

> **Critical & High Severity:**

Reviewer 2: ```suggestion
**Critical and high severity:**
```

- Organization security updates (SAML, 2FA, OAuth restrictions)
- Protected branch policy overrides or deletions
- Repository access and visibility changes
- Secret scanning alerts
- Organization member updates

> **Medium & Low Severity:**

Reviewer 2: ```suggestion
**Medium and low severity:**
```

- Repository member updates
- Organization moderator additions
- Integration installations

> **Advanced Security Detections:**

Reviewer 2: The bold text is going to render in the same paragraph as the text below it. Is that the intention? If we want this to be a heading, we should start it with a `####` for semantic HTML.

Author: Ty. Updated this

The system includes 26 sub-types of GitHub advanced security change detections that monitor for modifications to security features like Dependabot alerts.

> ### Okta Detections

Reviewer 2: ```suggestion
### Okta detections
```

Track identity and access management events in your Okta environment:

> **High Severity:**

Reviewer 2: ```suggestion
**High severity:**
```

- Admin MFA disabled
- All MFA factors reset
- OAuth token reuse patterns
- Rate limit violations
- Sign-on policy evaluations
- IDP lifecycle changes
- API token creation/revocation
- Security threat events

> **Medium Severity:**

Reviewer 2: ```suggestion
**Medium severity:**
```

- Excessive MFA failures
- Multiple password reset attempts
- Dormant account access
- Support-initiated password or MFA resets

> ### Teleport Detections

Reviewer 2: ```suggestion
### Teleport detections
```

Monitor Teleport-specific security events:

> **High Severity:**

Reviewer 2: ```suggestion
**High severity:**
```

- Root SSH session initiation
- Authentication without MFA for local accounts
- Unusual authentication failure patterns
- Role creation/updates/deletions
- Connector updates
- Unusual session commands

> ### Cross-Platform Detections

Reviewer 2: ```suggestion
### Cross-platform detections
```

> **Impossible Travel Detection** (GitHub, Okta, Teleport):

Reviewer 2: ```suggestion
**Impossible travel detection (GitHub, Okta, Teleport):**
```
For consistency with other bold text used as a list heading

- Severity: High
- Triggers when login attempts occur from multiple geographic locations within a timeframe that makes legitimate travel physically impossible

> ## Alert Management

Reviewer 2: ```suggestion
## Alert management
```

> 

Reviewer 2: What does alert management mean in this context? I would add a short paragraph explaining what this section is about.

Author: removed this section

### Current Capabilities
- View alerts across all integrated systems
- Filter by severity level

> - Investigate related events

Reviewer 2: I would expand each of these into a sentence. Otherwise, I'm not sure what each item means in terms of what I can do with the Web UI.

### Planned Features
- Alert resolution workflows
- Acknowledgment and muting capabilities
- Forwarding alerts to external services 

> - Custom detection creation

Reviewer 2: We don't typically publish roadmap items in the docs outside of the Upcoming Releases page. Would it make sense to add this there or remove it?

Author: I've removed this

> ## Frequently Asked Questions

Reviewer 2: ```suggestion
## Frequently asked questions
```

### Can I write my own detections?
No, detections are currently pre-configured per integration and cannot be customized.

### Can I resolve an alert?
Not currently, but we plan to add workflows to resolve, acknowledge, and mute alert types in future updates.

### What determines the severity level?
Severity levels are defined by the Identity Security Team based on the potential security impact. Feedback on severity assignments is welcome.

### Can I forward these events to another service?
This feature is planned for future release.

### How can I request new detection types?
For new detection requests based on customer needs, reach out to the Identity Security Team. Customer feedback is welcomed and helps prioritize new detection development.

> ## Next Steps

Reviewer 2: ```suggestion
## Next steps
```
